inbm/dispatcher-agent/dispatcher/aota/application_command.py

```diff
@@ -52,6 +52,38 @@ def __init__(self,
 
     def verify_command(self, cmd: str) -> None:
         check_application_command_supported(cmd)
+    
+    def run_signature_check(self, path_to_file: str) -> None:
+        """
+        Runs the signature check on the given file to authenticate the OTA update package.
+
+        This method checks if the device is provisioned with an OTA package check certificate.
+        If the device is provisioned and a signature is provided, it will verify the signature
+        of the file using the specified hash algorithm. If the signature verification fails or
+        if a signature is not provided while the device is provisioned, it raises an AotaError.
+        
+        If the device is not provisioned for signature verification, a warning is logged, and
+        the signature check is skipped.
+
+        @param path_to_file: The file path to the OTA update package that needs to be verified.
+        
+        @return: None. It may raise an AotaError if a required signature is not provided or
+        if verification fails.
+        """
+        if os.path.exists(OTA_PACKAGE_CERT_PATH):
+            if self._signature is not None:
```

Comment:
Add error handling for the case when `self._signature` is `None` but `self._hash_algorithm` is not, which could indicate a manifest error. [medium]

Reply:
this is handled in another class